### PR TITLE
fix(components): [table] add tabindex to tbody to fix #13991

### LINKED ISSUE #13991: [Component] [table] el-dialog prevents el-table from scrolling left and right via ← and →
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.9`
- Browser / OS: `Edg/115.0.1901.203 / MacOS 13.4.1`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHJlYWN0aXZlLCB2ZXJzaW9uIGFzIHZ1ZVZlcnNpb24gfSBmcm9tICd2dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5cbmNvbnN0IG1zZyA9IHJlZignSGVsbG8gV29ybGQhJylcbmNvbnN0IHZpc2libGUgPSByZWYodHJ1ZSlcbmNvbnN0IGl0ZW1zID0gcmVhY3RpdmUoW1xuICB7aWQ6IDEsIG5hbWU6ICdoaSBoaSBoaSBoaSBoaSd9LFxuICB7aWQ6IDIsIG5hbWU6ICdobyBobyBobyBobyBobyd9LFxuICB7aWQ6IDMsIG5hbWU6ICdoYSBoYSBoYSBoYSBoYSd9LFxuXSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxoMT57eyBtc2cgfX08L2gxPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJpdGVtc1wiPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwiaWRcIiBsYWJlbD1cImlkXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWUxXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWUyXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWUzXCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWU0XCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWU1XCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWU2XCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gcHJvcD1cIm5hbWVcIiBsYWJlbD1cIm5hbWU3XCIgLz5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJhY3Rpb25zXCIgZml4ZWQ9XCJyaWdodFwiIHdpZHRoPVwiMTAwXCI+XG4gICAgICAgIDxidXR0b24+ZG9udCBjbGljazwvYnV0dG9uPiBcbiAgICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICAgIDwvZWwtdGFibGU+XG4gIDxlbC1kaWFsb2cgdi1tb2RlbD1cInZpc2libGVcIj5cbiAgICA8ZWwtdGFibGUgOmRhdGE9XCJpdGVtc1wiPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTFcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTJcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTNcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTRcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTVcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTZcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wPVwibmFtZVwiIGxhYmVsPVwibmFtZTdcIiAvPlxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cImFjdGlvbnNcIiBmaXhlZD1cInJpZ2h0XCIgd2lkdGg9XCIxMDBcIj5cbiAgICAgICAgPGJ1dHRvbj5kb250IGNsaWNrPC9idXR0b24+IFxuICAgICAgPC9lbC10YWJsZS1jb2x1bW4+XG4gICAgPC9lbC10YWJsZT5cbiAgPC9lbC1kaWFsb2c+XG48L3RlbXBsYXRlPlxuXG48c3R5bGUgc2NvcGVkPlxuYnV0dG9ue1xuICB3aGl0ZS1zcGFjZTogbm93cmFwO1xufVxuOjp2LWRlZXAoLmNlbGwpe1xuICB3aGl0ZS1zcGFjZTogbm93cmFwICFpbXBvcnRhbnQ7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

1. click the table in the dialog, press ← or → to scroll the table (it does not work!)
2. close the dialog, click the table, press ← or → to scroll the table (it works)

### What is Expected?

Both works

### What is actually happening?

One does not work, the other works

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by FrankFang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 681e5ca</samp>

Fixed the unwanted outline of the table body when focused by setting `tabIndex` to -1 and removing the outline style. This improved the design and accessibility of the `table` component.

## Related Issue

Fixes #13991.

### <samp>🤖 Generated by Copilot at 681e5ca</samp>

* Remove outline from table body when focused ([link](https://github.com/element-plus/element-plus/pull/14020/files?diff=unified&w=0#diff-9ea53ccf31f5ae3be420af1491bff7d8883a1f156b8407db6c87c0220b24cc3fL67-R67), [link](https://github.com/element-plus/element-plus/pull/14020/files?diff=unified&w=0#diff-50fcce7b86d8998ca0385dda1351e7376e5fa09b4939adec02eb34dda9e32ba5R37-R42))

### CODE REVIEW

File: packages/theme-chalk/src/table.scss [hunk 1]
```diff
@@ -34,6 +34,12 @@
     }
   }
 
+  tbody {
+    &:focus {
+      outline: none;
+    }
+  }
+
```
Comment (btea) [inline packages/theme-chalk/src/table.scss:42]: The style here seems unnecessary to add.

**Review by btea — APPROVED** [review]

File: packages/components/table/src/table-body/index.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ export default defineComponent({
   render() {
     const { wrappedRowRender, store } = this
     const data = store.states.data.value || []
-    return h('tbody', {}, [
+    return h('tbody', { tabIndex: -1 }, [
```
Comment (HeftyKoo) [inline packages/components/table/src/table-body/index.ts:67]: Thanks, would you mind adding some comments for this code?

File: packages/theme-chalk/src/table.scss (hunk 1 above)
Comment (btea) [reply packages/theme-chalk/src/table.scss]: Sorry, this should not be removed directly, but override the `outline` style of the `focus-visible` pseudo-class.

image

Comment (tolking) [reply packages/theme-chalk/src/table.scss]: agree @FrankFang

File: packages/components/table/src/table-body/index.ts (hunk 2 above)
Comment (tolking) [reply packages/components/table/src/table-body/index.ts:67]: It is used to capture focus events triggered by the mouse

**Review by HeftyKoo — APPROVED** [review]

File: packages/theme-chalk/src/table.scss (hunk 1 above)
Comment (FrankFang) [reply packages/theme-chalk/src/table.scss]: Any suggestions? I am not sure whether `outline: none;` affects the other components' styles.
But removing `outline: none` makes the table component weird.
For now, I just add `outline: none` back.

Comment (btea) [reply packages/theme-chalk/src/table.scss]: You should only need to add the following styles: 
```scss
tbody {
  &:focus-visible {
    outline: none;
  }
}
```

Comment (FrankFang) [reply packages/theme-chalk/src/table.scss:42]: It makes sense. I've pushed my code.

File: packages/components/table/src/table-body/index.ts (hunk 2 above)
Comment (FrankFang) [reply packages/components/table/src/table-body/index.ts:67]: Ok, I will add the link from MDN

**Comment by btea** [comment]
@FrankFang Thanks!

**Pull request merged**